### PR TITLE
fix: update country detector

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### Issue

Current implementation of how we detect the country doesn't work as expected. As it's picking up the country where vercel is currently hosted compared to where the user actually is

### Solution

Remove the use of middleware and create an api endpoint that picks up the geolocation of the user instead

### CODE REVIEW

File: libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.spec.tsx [hunk 1]
```diff
@@ -9,6 +9,16 @@ import { languageRefinements } from '../../data'
...
 import { NEXT_COUNTRY } from './data'
 
+global.fetch = jest.fn(
+  async () =>
+    await Promise.resolve({
+      json: async () =>
+        await Promise.resolve({
+          country: 'US'
+        })
+    })
```
Comment (scottkelso) [inline libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.spec.tsx:19]: Would be super slick to have this refactored into that data.ts file.

File: libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx [hunk 2]
```diff
@@ -52,27 +52,21 @@ export function CountryLanguageSelector({
...
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    void fetch('/api/jf/watch.html/geolocation').then((response) => {
+      void response.json().then((data: { country?: string }) => {
+        if (data.country != null) {
+          const countryName = parseCountryName(data.country)
+          setCountry(countryName)
+          setCountryCode(data.country)
+        }
+      })
```
Comment (scottkelso) [inline libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx:68]: Can this be done with `await` keywords instead?

**Review by scottkelso — CHANGES_REQUESTED** [review]
Looking good! Left just 2 things to check.

File: libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx [hunk 3]
```diff
@@ -52,31 +52,25 @@ export function CountryLanguageSelector({
...
+  const findUserCountry = useCallback(async () => {
+    await fetch('/api/jf/watch.html/geolocation').then(async (response) => {
+      await response.json().then((data: { country?: string }) => {
+        if (data.country != null) {
+          const countryName = parseCountryName(data.country)
+          setCountry(countryName)
+          setCountryCode(data.country)
+        }
+      })
+    })
```
Comment (scottkelso) [inline libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx:69]: Was thinking more can it be done like...
```suggestion
    const response = await fetch('/api/jf/watch.html/geolocation');
    const data: { country?: string } = await response.json();
    
    if (data.country) {
      const countryName = parseCountryName(data.country);
      setCountry(countryName);
      setCountryCode(data.country);
    }
```

**Review by scottkelso — CHANGES_REQUESTED** [review]
Nice refactor - one one thing I think can be improved.

**Review by scottkelso — APPROVED** [review]

File: apps/watch/pages/api/jf/watch.html/geolocation.ts
Comment (tataihono) [inline apps/watch/pages/api/jf/watch.html/geolocation.ts:1]: can this move to apps/watch/pages/api/geolocation.ts

**Pull request merged**